Celeste.Mod.mm/Patches/OuiOptions.cs

```diff
@@ -0,0 +1,21 @@
+﻿#pragma warning disable CS0626 // Method, operator, or accessor is marked external and has no attributes on it
+
+using Celeste.Mod;
+using Celeste.Mod.Helpers;
+using MonoMod;
+using System.Collections;
+
+namespace Celeste {
+    class patch_OuiOptions : OuiOptions {
+        public extern IEnumerator orig_Leave(Oui next);
+        public override IEnumerator Leave(Oui next) {
```

Comment:
Re: discussion in Discord - this only gets called from the main menu options. If you want settings to apply after changes from the in-game menu, you'd probably want to add this in `UserIO` instead